internal/common/compilation.bzl

```diff
@@ -329,3 +365,12 @@ def ts_providers_dict_to_struct(d):
     if type(value) == type({}):
       d[key] = struct(**value)
   return struct(**d)
+
+def collect_js(ctx, deps):
```

Comment:
This is extracted from rules_closure. What would you think about moving this around in the rules_closure repo to make it an importable function?

Reply:
Its heavily modified from the rules_closure implementation (and the rules closure implementation was within a directory named private). Also I'm not sure that its completely wise to make rules_typescript directly depend upon rules_closure since it'll cause user's of these rules to have to clone an entire repo they won't necessarily use

Reply:
+1 for copying in this case

Reply:
Done.